controllers/dnspolicy_status.go

```diff
@@ -90,7 +92,15 @@ func (r *DNSPolicyReconciler) calculateStatus(ctx context.Context, dnsPolicy *v1
 
 	meta.SetStatusCondition(&newStatus.Conditions, *enforcedCondition)
 
-	propagateRecordConditions(recordsList, newStatus)
+	for _, record := range recordsList.Items {
+		for _, reference := range record.GetOwnerReferences() {
+			if reference.Controller != nil && *reference.Controller && reference.Name == dnsPolicy.Name && reference.UID == dnsPolicy.UID {
+				controlledRecords.Items = append(controlledRecords.Items, record)
+			}
+		}
```

Comment:
What's the significance of it being a controller reference or not? This also differs from how we find related sub resources [elsewhere](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/dnspolicy_dnsrecords.go#L167), wondering which approach is more desirable, not sure if we use the owner ref in other polices or not.

Reply:
I just copied this logic from [here](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/dnspolicy_status.go#L103), happy to change it, if there's a better pattern, this does seem to work for me though.